# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLShaderValidator.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLShaderValidator.h
--- a/dom/canvas/WebGLShaderValidator.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLShaderValidator.h	Wed Jan 08 22:19:16 2020 +0000
@@ -7,6 +7,7 @@
 #define WEBGL_SHADER_VALIDATOR_H_
 
 #include <string>
+#include <unordered_map>
 
 #include "GLDefs.h"
 #include "GLSLANG/ShaderLang.h"
@@ -30,13 +31,12 @@
   std::vector<sh::Uniform> mUniforms;
   std::vector<sh::Varying> mVaryings;
 
+  std::unordered_map<std::string, std::string> mNameMap;
+
   int mMaxVaryingVectors = 0;
 
   bool CanLinkTo(const ShaderValidatorResults& vert,
                  nsCString* const out_log) const;
-  bool FindUniformByMappedName(const std::string& mappedName,
-                               std::string* const out_userName,
-                               bool* const out_isArray) const;
   size_t SizeOfIncludingThis(MallocSizeOf) const;
 };
 